src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,307 @@
+            /// </summary>
+            public override int Read(byte[] buffer, int offset, int count)
+            {
+                throw new NotSupportedException();
+            }
+
+            /// <summary>
+            /// Sets the position within the stream. This operation is not supported by this stream.
+            /// </summary>
+            public override long Seek(long offset, SeekOrigin origin)
+            {
+                throw new NotSupportedException();
+            }
+
+            /// <summary>
+            /// Gets the length in bytes of the stream. This is not supported by this stream.
+            /// </summary>
+            public override void SetLength(long value)
+            {
+                throw new NotSupportedException();
+            }
+
+            /// <summary>
+            /// Writes a sequence of bytes to the current stream.
+            /// </summary>
+            /// <param name="buffer">An array of bytes. This method copies count bytes from buffer to the current stream.</param>
+            /// <param name="offset">The zero-based byte offset in buffer at which to begin copying bytes to the current stream.</param>
+            /// <param name="count">The number of bytes to be written to the current stream.</param>
+            public override void Write(byte[] buffer, int offset, int count)
+            {
+                Span<byte> value = buffer.AsSpan().Slice(offset, count);
+
+                this.StartWritingByteValueInChunks(value);
+            }
+
+            /// <summary>
+            /// Asynchronously writes a portion of a byte array to the underlying stream in chunks.
+            /// </summary>
+            /// <param name="buffer">The byte array from which data will be written.</param>
+            /// <param name="offset">The zero-based byte offset in the buffer at which to begin copying bytes to the stream.</param>
+            /// <param name="count">The maximum number of bytes to write.</param>
+            /// <param name="token">A CancellationToken to observe while waiting for the task to complete.</param>
+            /// <returns>A task representing the asynchronous write operation.</returns>
+            public override async Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken token)
+            {
+                ReadOnlyMemory<byte> value = buffer.AsMemory().Slice(offset, count);
+                await this.StartWritingByteValueInChunksAsync(value);
+            }
+
+            /// <summary>
+            /// Writes the byte value represented by the provided read-only span in chunks using Base64 encoding.
+            /// </summary>
+            /// <param name="value">The read-only span containing the byte value to be written.</param>
+            private void StartWritingByteValueInChunks(ReadOnlySpan<byte> value)
+            {
+                // Process the input value in chunks
+                for (int i = 0; i < value.Length; i += chunkSize)

```

Comment:
Where is `chunkSize` defined?

Reply:
Check here: https://github.com/OData/odata.net/pull/2880#discussion_r1517230380